corehq/apps/reports/commtrack/data_sources.py

```diff
@@ -418,26 +432,30 @@ def get_data(self):
             for spoint_id, loc_id in spoint_loc_map.items():
                 loc = locations[loc_id]
 
-                form_ids = StockReport.objects.filter(
+                results = StockReport.objects.filter(
                     stocktransaction__case_id=spoint_id
-                ).exclude(
-                    date__lte=self.start_date
-                ).exclude(
-                    date__gte=self.end_date
                 ).values_list(
-                    'form_id', flat=True
+                    'form_id',
+                    'date'
                 ).order_by('-date').distinct()  # not truly distinct due to ordering
+
                 matched = False
-                for form_id in form_ids:
+                for form_id, date in results:
+                    if self.converted_start_datetime > date:
+                        break
+
                     try:
-                        if XFormInstance.get(form_id).xmlns in form_xmlnses:
+                        if self.converted_end_datetime >= date and \
+                           XFormInstance.get(form_id).xmlns in form_xmlnses:
```

Comment:
(not sure whether this is actionable since it looks like it's ~the same behavior, just noting)